### PR TITLE
Improve date calculation, refs #12082

### PR DESCRIPTION
**Pull request by mcantelon**
* Added logic so event types are derived from the types present in descendant
  information object's events

* Removed automatic selection of single events (so their dates can be replaced)
  given there is now an option to add a new event with dates calculated from
  descendant dates

* Changed logic determining whether a date calculation link should be shown so
  it's not shown if no events exist in descendent events

### CODE REVIEW

**Comment by jraddaoui** [comment]
Hi @mcantelon!

I can see the commit body formatting issue in here too. It looks to be a "Github" thing and only in some places ...

If you click in the three dots on your commit above (without leaving this page), it looks like Github is taking in consideration the 72 chars rule. However, if you click in the commit link and go to the commit page, the body format is maintained. As you can see, it's a silly thing, I was just wondering how it happened. That being said, if we could keep the commit body lines <= than 72 chars, that would be great.

Reviewing the code now ;)

File: apps/qubit/modules/informationobject/actions/calculateDatesAction.class.php [hunk 1]
```diff
@@ -115,19 +106,34 @@ public function execute($request)
...
     {
-      $message = $this->i18n->__('Warning: Selected date range for the specified event will be overwritten.');
+      $message = $this->i18n->__("Warning: If selecting an event, rather than an event type, the selected event's date range will be overwritten.");
       $this->getUser()->setFlash('notice', $message);
     }
   }
 
+  protected function validCalculationSelectionCheck()
+  {
+    return $this->eventId xor $this->eventTypeId;
```
Comment (jraddaoui) [inline apps/qubit/modules/informationobject/actions/calculateDatesAction.class.php:134]: :clap:

**Review by jraddaoui — APPROVED** [review]
Nicely done @mcantelon!

**Comment by mcantelon** [comment]
Ah cool, I'll make the commit lines <= 72 chars! Thanks @jraddaoui!

**Pull request merged**